test/cppsuite/tests/schema_sweep.cpp

```diff
@@ -0,0 +1,147 @@
+class schema_sweep : public test {
+public:
+    schema_sweep(const test_args &args) : test(args)
+    {
+        init_operation_tracker();
+    }
+
+    // Reconfigures the connection with different sweep server parameters.
+    void
+    custom_operation(thread_worker *tw) override final
+    {
+        WT_CONNECTION *conn = connection_manager::instance().get_connection();
+        bool aggressive_sweep = false;
+        const std::string aggressive_sweep_cfg =
+          "file_manager=(close_handle_minimum=0,close_idle_time=1,close_scan_interval=1)";
+        const std::string default_sweep_cfg =
+          "file_manager=(close_handle_minimum=250,close_idle_time=30,close_scan_interval=10)";
+
+        while (tw->running()) {
+            tw->sleep();
+            testutil_check(conn->reconfigure(
+              conn, aggressive_sweep ? default_sweep_cfg.c_str() : aggressive_sweep_cfg.c_str()));
+            aggressive_sweep = !aggressive_sweep;
+        }
+    }
+
+    // Keeps creating collections.
+    void
+    insert_operation(thread_worker *tw) override final
+    {
+        auto max_coll_count = 1000;
+        while (tw->running()) {
+            if (tw->db.get_collection_count() < max_coll_count) {
+                auto session = connection_manager::instance().create_session();
+                tw->db.add_collection(session);
+            }
+            tw->sleep();
+        }
+    }
+
+    // Keeps deleting collections.
+    void
+    remove_operation(thread_worker *tw) override final
+    {
+        while (tw->running()) {
+            if (tw->db.get_collection_count() != 0) {
+                std::string cfg = "force=";
+                cfg += random_generator::instance().generate_bool() ? "true" : "false";
+                // We don't need to check whether the collection has been actually removed or not.
+                tw->db.remove_random_collection(cfg);
+            }
+            tw->sleep();
+        }
+    }
+
+    // Selects a random collection and performs an update on it. Note that the collection can be
+    // deleted while the update is happening.
+    void
+    update_operation(thread_worker *tw) override final
+    {
+        while (tw->running()) {
+            tw->sleep();
+            {
+                /*
+                 * Use a function that holds a lock while retrieving the name. If we retrieve the
+                 * collection then get the name from it, another thread may free the data allocated
+                 * to the collection in parallel since we don't hold a lock on the database here.
+                 * The name is empty if the database is empty.
+                 */
+                const std::string collection_name(tw->db.get_random_collection_name());
+                if (collection_name.empty())
+                    continue;
+                auto session = connection_manager::instance().create_session();
+
+                WT_CURSOR *cursor;
```

Comment:
I think we want a `scoped_cursor` instead of `WT_CURSOR` here

Reply:
It would be great but we cannot.
The way `scoped_cursor`s are implemented, we fail if they cannot be opened, see `scoped_cursor::reinit`. In this test, it is possible to fail when opening the cursor if the collection is removed at the same time. In order to handle this error, I need a WiredTiger cursor.